src/pmhn/_trees/_simulate.py

```diff
@@ -136,6 +136,9 @@ def _simulate_tree(
                     ):
                         exit_while = True
                         break
+                # delete Node, since it is not needed
+                else:
+                    new_node.parent = None
```

Comment:
The _simulate_tree function returns a dictionary that contains the nodes of the generated tree with their corresponding waiting times. However, the only thing we need is the root node (which is contained in the first node-waiting time pair of the returned dictionary), because we have access to the other nodes e.g by calling root_node.descendants. If a node is not added to the dictionary (because the waiting time is larger than the sampling time), we still need to delete the node. Otherwise, the node would be still contained in the tree (though not in the returned dictionary). Initially, I thought that this was a big problem for the comparison plots (mutation frequencies, tree size distribution), but I verified that when I wrote trees to a CSV-file, I only considered the nodes that were in the dictionary.

In the generate_data rule (treemhn-bayes.smk) you probably saw following lines:
trees = []
for i, tree in enumerate(trees_dict):
            for key, val in tree.items():
                print(RenderTree(key))
                tree_wrapper =TreeWrapperCode(key) 
                trees.append(tree_wrapper)
                break 
Here, I apply TreeWrapperCode to the node of the first node-waiting time pair of each dictionary (one dictionary for each tree) and append it to the trees list. This wouldn't have worked if there hadn't been the line 
else:
       new_node.parent = None
in the _simulate_tree function. 

Since we don't need the waiting times of each node, maybe I should change the _simulate_tree function such that it only returns the root node, what do you think?

Reply:
I see, thanks for explanation! I like the idea to update `_simulate_tree`, but for another PR – I'd suggest you to merge this one on Friday :slightly_smiling_face: